AirCasting/MobilePeripheralSessionManager.swift

```diff
@@ -0,0 +1,73 @@
+// Created by Lunar on 07/06/2021.
+//
+
+import Foundation
+import CoreBluetooth
+
+class MobilePeripheralSessionManager {
+    private let measurementStreamStorage: MeasurementStreamStorage
+    private lazy var locationProvider = LocationProvider()
+    
+    var activeMobileSession: MobileSession?
+    var streamsIDs: [String: MeasurementStreamLocalID] = [:]
+    
+    init(measurementStreamStorage: MeasurementStreamStorage) {
+        self.measurementStreamStorage = measurementStreamStorage
+    }
+    
+    func startRecording(session: Session, peripheral: CBPeripheral) throws {
+        try measurementStreamStorage.createSession(session)
+        try measurementStreamStorage.updateSessionStatus(.NEW, for: session.uuid)
+        activeMobileSession = MobileSession(peripheral: peripheral, session: session)
+    }
+    
+    func handlePeripheralMeasurement(_ measurement: PeripheralMeasurement) {
+        if activeMobileSession == nil {
+            return
+        }
+        if activeMobileSession?.peripheral == measurement.peripheral {
```

Comment:
I kept it separate for readability because we want to check both cases: "And inside this method is where we check whether there is an active mobile session and whether the peripheral which sent that message is associated with this mobile session.". However if you think it's a mistake I can change it.

Reply:
Just a redundancy. To me it looks a bit verbose, but If you think this makes stuff easier to read I think it's ok to leave it as is.